packages/core/src/image_performance_warning.ts

```diff
@@ -0,0 +1,184 @@
+const SCAN_DELAY = 200;
+
+const OVERSIZED_IMAGE_TOLERANCE = 1200;
+
+type ImagePerformanceAlertOptions = {
+  disableImageSizeWarning?: boolean,
+  disableImageLazyLoadWarning?: boolean
+};
+
+@Injectable({providedIn: 'root'})
+export class ImagePerformanceWarning implements OnDestroy {
+  // Map of full image URLs -> original `ngSrc` values.
+  private window: Window|null = null;
+  private observer: PerformanceObserver|null = null;
+  private options: ImagePerformanceAlertOptions = processConfig(inject(IMAGE_CONFIG));
+  private lcpImageUrl?: string;
+
+  constructor() {}
+
+  public start() {
+    if (typeof IntersectionObserver === 'undefined') {
+      return;
+    }
+    this.observer = this.initPerformanceObserver();
+    const win = getDocument().defaultView;
+    if (typeof win !== 'undefined') {
+      this.window = win;
+      // Wait to avoid race conditions where LCP image triggers
+      // load event before it's recorded by the performance observer
+      const waitToScan = () => {
+        setTimeout(this.scanImages.bind(this), SCAN_DELAY);
+      };
+      this.window?.addEventListener('load', waitToScan);
+    }
+  }
+
+  ngOnDestroy() {
+    if (!this.observer) return;
+    this.observer?.disconnect();
+  }
+
+  private initPerformanceObserver(): PerformanceObserver|null {
+    if (typeof IntersectionObserver === 'undefined') {
+      return null;
+    }
+    const logLcpImage = (imageUrl: string) => {
+      this.lcpImageUrl = imageUrl;
+    };
+    const observer = new PerformanceObserver((entryList) => {
+      const entries = entryList.getEntries();
+      if (entries.length === 0) return;
+      // We use the latest entry produced by the `PerformanceObserver` as the best
+      // signal on which element is actually an LCP one. As an example, the first image to load on
+      // a page, by virtue of being the only thing on the page so far, is often a LCP candidate
+      // and gets reported by PerformanceObserver, but isn't necessarily the LCP element.
+      const lcpElement = entries[entries.length - 1];
+
+      // Cast to `any` due to missing `element` on the `LargestContentfulPaint` type of entry.
+      // See https://developer.mozilla.org/en-US/docs/Web/API/LargestContentfulPaint
+      const imgSrc = (lcpElement as any).element?.src ?? '';
+
+      // Exclude `data:` and `blob:` URLs, since they are fetched resources.
+      if (imgSrc.startsWith('data:') || imgSrc.startsWith('blob:')) return;
+
+      logLcpImage(imgSrc);
+    });
+    observer.observe({type: 'largest-contentful-paint', buffered: true});
+    return observer;
+  }
+
+  private scanImages(): void {
+    const images = getDocument().querySelectorAll('img');
+    let lcpElementFound, lcpElementLoadedCorrectly = false;
+    images.forEach(image => {
+      if (!this.options?.disableImageSizeWarning) {
+        for (const image of images) {
+          if (this.isOversized(image) && !image.getAttribute('ng-img')) {
```

Comment:
We can add a comment here to mention that we exclude images with NgOptimizedImage directive from this check (e.g. we check for `ng-img` attribute). I'd also propose flipping the operands, since the `isOversized` does some work that we can avoid:
```suggestion
          if (!image.getAttribute('ng-img') && this.isOversized(image)) {
```

Reply:
done